docs/recipes/accounts.md

```diff
@@ -38,5 +38,75 @@ the amount of data you intend to store in the account.
   </CodeGroupItem>
 </CodeGroup>
 
+## Program Derived Address
+
+[Prgoram dervied address (PDA)][3] is like a normal address but
+
+1. falling off ed25519 curve
+2. using program to sign instead of private key
+
+::: tip
+Although PDA is derived by a program id, it doesn't means the PDA is owned by the same program. (Take an example, you can initialize your PDA as a token account which is an account owned by token program)
+:::
+
+### Generate a PDA
+
+To calculate a PDA, you need
+
+1. seed
+2. program id
+
+There are two ways you can calculate a PDA
+
+1. Create Program Address
+
+It may fail because the result (pda) is on curve. You can use `findProgramAddress` to reserve your meaningful seed.
+
+<CodeGroup>
+  <CodeGroupItem title="TS" active>
+
+@[code](@/code/accounts/program-derived-address/derived-a-pda/create-program-address.en.ts)
+
+  </CodeGroupItem>
+</CodeGroup>
+
+2. Find Program Address
+
+`findProgramAddress` will add a extra byte in the end of your seed. It starts from 255 to 0 and return the first off-curve public key. You always get the same result if your pass the same program id and seed.
+
+<CodeGroup>
+  <CodeGroupItem title="TS" active>
+
+@[code](@/code/accounts/program-derived-address/derived-a-pda/find-program-address.en.ts)
+
+  </CodeGroupItem>
+</CodeGroup>
+
+### Sign a PDA
+
+We can use `invoke_signed` to sign a PDA.
+
+#### Program
```

Comment:
We probably can just explain the program vs client here, then list the client and program in separate code blocks.

So something like
`PDAs can only be signed for within the program. Below is a program example of signing with a PDA and calling the program with the client` and no other explanation

Reply:
fixed at [here](https://github.com/solana-dev-adv/solana-cookbook/pull/37/commits/814ffc3dbf81f54b507fa45c416cf63121573ed4)